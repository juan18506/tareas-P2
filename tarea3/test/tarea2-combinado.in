crearFecha 1/2/2023
imprimirFecha
copiarFecha
aumentarDias 28
copiarFecha
liberarFecha

# año bisiesto
crearFecha 1/2/2024
imprimirFecha
aumentarDias 28
liberarFecha

# mas de un mes
crearFecha 18/12/2023
imprimirFecha
aumentarDias 50
liberarFecha

# mas de un mes y bisiesto
crearFecha 18/12/2023
imprimirFecha
copiarFecha
aumentarDias 73
copiarFecha
liberarFecha

# mas de un año
crearFecha 18/12/2022
imprimirFecha
aumentarDias 366
liberarFecha

# mas de un año y bisiesto
crearFecha 18/12/2023
imprimirFecha
aumentarDias 366
liberarFecha

crearFecha 22/3/2023
crearEvento 1 Descripcion 1
copiarEvento
imprimirFechaEvento
posponerEvento 30
imprimirFechaEvento
posponerEvento 10
imprimirEvento
copiarEvento
liberarEvento

# año bisiesto 

crearFecha 01/02/2024
crearEvento 2 Descripcion 2
imprimirFechaEvento
posponerEvento 28
imprimirFechaEvento
copiarEvento
posponerEvento 1
imprimirEvento
copiarEvento
liberarEvento

crearAgenda

estaEnAgenda 1

crearFecha 19/4/2023
crearEvento 1 Descripcion uno
agregarEnAgenda

estaEnAgenda 1

crearFecha 15/4/2023
crearEvento 2 Descripcion dos
agregarEnAgenda

crearFecha 21/4/2023
crearEvento 3 Descripcion tres
agregarEnAgenda

crearFecha 31/12/2022
crearEvento 4 Descripcion cuatro
agregarEnAgenda

imprimirAgenda

posponerEnAgenda 4 109

imprimirAgenda

hayEventosFechaAgenda 31/12/2022

imprimirEventosFechaAgenda 19/4/2023

removerDeAgenda 1

imprimirCopiaAgenda

liberarAgenda

crearAgenda

crearPersona 1 20 Persona1

imprimirCopiaPersona

# creo cuatro eventos para ingresarlos a la agenda de la persona

crearFecha 22/3/2023
crearEvento 1 Descripcion uno
agregarEventoPersona

crearFecha 23/3/2023
crearEvento 2 Descripcion dos
agregarEventoPersona

imprimirCopiaPersona

crearFecha 24/3/2023
crearEvento 3 Descripcion tres
agregarEventoPersona

crearFecha 1/4/2023
crearEvento 4 Descripcion cuatro
agregarEventoPersona

imprimirCopiaPersona

# chequéo si existen diferentes ids de eventos y los obtengo

estaEnAgendaPersona 3
estaEnAgendaPersona 5
estaEnAgendaPersona 4
estaEnAgendaPersona 0

imprimirEventoPersona 3
imprimirEventoPersona 1

liberarPersona

crearArbolPersonas

crearAgenda
crearPersona 1 1 Persona1
crearFecha 1/1/1111
crearEvento 1 Descripcion uno
agregarEventoPersona
insertarPersonaArbolPersonas

imprimirArbolPersonas

removerArbolPersonas 1

estaEnArbolPersonas 1
esVacioArobolPersonas

crearAgenda
crearPersona 5 55 Persona5
crearFecha 5/5/55
crearEvento 5 Descripcion5
agregarEventoPersona
insertarPersonaArbolPersonas

crearAgenda
crearPersona 10 10 Persona10
insertarPersonaArbolPersonas

crearAgenda
crearPersona 20 20 Persona20
crearFecha 2/2/20
crearEvento 20 Descripcion20
agregarEventoPersona
insertarPersonaArbolPersonas

crearAgenda
crearPersona 15 15 Persona15
crearFecha 5/5/15
crearEvento 15 Descripcion15
agregarEventoPersona
insertarPersonaArbolPersonas

imprimirCantidadArbolPersonas
alturaArbolPersonas
estaEnArbolPersonas 15
maxIdPersonaArbolPersonas

aplanarArbolPersonas

removerArbolPersonas 10
estaEnArbolPersonas 10

mayoresEdadArbolPersonas 10

removerArbolPersonas 20

esPerfectoArbolPersonas

crearAgenda
crearPersona 2 22 Persona2
crearFecha 2/2/22
crearEvento 2 Descripcion2
agregarEventoPersona
insertarPersonaArbolPersonas

esPerfectoArbolPersonas

liberarArbolPersonas

crearListaPersonas

crearAgenda
crearPersona 1 11 Persona1
crearFecha 1/11/11
crearEvento 1 Descripcion1
agregarEventoPersona
insertarPersonaListaPersonas 3

crearAgenda
crearPersona 2 22 Persona2
crearFecha 2/22/22
crearEvento 2 Descripcion2
agregarEventoPersona
insertarPersonaListaPersonas 2

crearAgenda
crearPersona 3 33 Persona3
crearFecha 3/33/33
crearEvento 3 Descripcion3
agregarEventoPersona
insertarPersonaListaPersonas 3

crearAgenda
crearPersona 0 01 Persona0
crearFecha 01/10/1100
crearEvento 0 Descripcion0
agregarEventoPersona
insertarPersonaListaPersonas 1

imprimirCantidadListaPersonas
imprimirListaPersonas

estaEnListaPersonas 3
estaEnListaPersonas 5
estaEnListaPersonas 4
estaEnListaPersonas 0

concatenarConVacioListaPersonas

eliminarInicioListaPersonas

imprimirCantidadListaPersonas
imprimirListaPersonas

eliminarFinalListaPersonas

imprimirPersonaListaPersonas 2

imprimirCantidadListaPersonas
imprimirListaPersonas

eliminarFinalListaPersonas
eliminarFinalListaPersonas

imprimirCantidadListaPersonas
imprimirListaPersonas

crearYConcatenarListaPersonas

imprimirListaPersonas
imprimirCantidadListaPersonas

liberarListaPersonas

Fin
