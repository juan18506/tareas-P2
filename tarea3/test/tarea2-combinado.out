1>2>1/2/2023
3>La fecha copiada es: 1/2/2023
4>La nueva fecha aplazada 28 dias es: 1/3/2023
5>La fecha copiada es: 1/3/2023
6>7>#  año bisiesto.
8>9>1/2/2024
10>La nueva fecha aplazada 28 dias es: 29/2/2024
11>12>#  mas de un mes.
13>14>18/12/2023
15>La nueva fecha aplazada 50 dias es: 6/2/2024
16>17>#  mas de un mes y bisiesto.
18>19>18/12/2023
20>La fecha copiada es: 18/12/2023
21>La nueva fecha aplazada 73 dias es: 29/2/2024
22>La fecha copiada es: 29/2/2024
23>24>#  mas de un año.
25>26>18/12/2022
27>La nueva fecha aplazada 366 dias es: 19/12/2023
28>29>#  mas de un año y bisiesto.
30>31>18/12/2023
32>La nueva fecha aplazada 366 dias es: 18/12/2024
33>34>35>36>El evento copiado es: Evento 1:  Descripcion 1
Fecha: 22/3/2023
37>La fecha del evento es: 22/3/2023
38>39>La fecha del evento es: 21/4/2023
40>41>Evento 1:  Descripcion 1
Fecha: 1/5/2023
42>El evento copiado es: Evento 1:  Descripcion 1
Fecha: 1/5/2023
43>44>#  año bisiesto .
45>46>47>La fecha del evento es: 1/2/2024
48>49>La fecha del evento es: 29/2/2024
50>El evento copiado es: Evento 2:  Descripcion 2
Fecha: 29/2/2024
51>52>Evento 2:  Descripcion 2
Fecha: 1/3/2024
53>El evento copiado es: Evento 2:  Descripcion 2
Fecha: 1/3/2024
54>55>56>El evento con id 1 NO está en la agenda.
57>58>59>60>El evento con id 1 está en la agenda.
61>62>63>64>65>66>67>68>69>70>Evento 4:  Descripcion cuatro
Fecha: 31/12/2022
Evento 2:  Descripcion dos
Fecha: 15/4/2023
Evento 1:  Descripcion uno
Fecha: 19/4/2023
Evento 3:  Descripcion tres
Fecha: 21/4/2023
71>La nueva fecha del evento 4 pospuesto 109 dias es: 19/4/2023
72>Evento 2:  Descripcion dos
Fecha: 15/4/2023
Evento 4:  Descripcion cuatro
Fecha: 19/4/2023
Evento 1:  Descripcion uno
Fecha: 19/4/2023
Evento 3:  Descripcion tres
Fecha: 21/4/2023
73>No se encontraron eventos en la fecha determinada.
74>Evento 4:  Descripcion cuatro
Fecha: 19/4/2023
Evento 1:  Descripcion uno
Fecha: 19/4/2023
75>El evento con id 1 se removió de la agenda.
76>Evento 2:  Descripcion dos
Fecha: 15/4/2023
Evento 4:  Descripcion cuatro
Fecha: 19/4/2023
Evento 3:  Descripcion tres
Fecha: 21/4/2023
77>78>79>80>Persona 1:  Persona1, 20 años
81>#  creo cuatro eventos para ingresarlos a la agenda de la persona.
82>83>84>85>86>87>88>Persona 1:  Persona1, 20 años
Evento 1:  Descripcion uno
Fecha: 22/3/2023
Evento 2:  Descripcion dos
Fecha: 23/3/2023
89>90>91>92>93>94>95>Persona 1:  Persona1, 20 años
Evento 1:  Descripcion uno
Fecha: 22/3/2023
Evento 2:  Descripcion dos
Fecha: 23/3/2023
Evento 3:  Descripcion tres
Fecha: 24/3/2023
Evento 4:  Descripcion cuatro
Fecha: 1/4/2023
96>#  chequéo si existen diferentes ids de eventos y los obtengo.
97>El evento con id 3 está en la agenda de la persona.
98>El evento con id 5 NO está en la agenda de la persona.
99>El evento con id 4 está en la agenda de la persona.
100>El evento con id 0 NO está en la agenda de la persona.
101>Evento 3:  Descripcion tres
Fecha: 24/3/2023
102>Evento 1:  Descripcion uno
Fecha: 22/3/2023
103>104>105>106>107>108>109>110>111>Persona 1:  Persona1, 1 años
Evento 1:  Descripcion uno
Fecha: 1/1/1111
112>113>La persona con id 1 NO está en el árbol de personas.
114>El árbol de personas está vacío.
115>116>117>118>119>120>121>122>123>124>125>126>127>128>129>130>131>132>133>134>135>136>La cantidad de personas en el árbol es: 4.
137>La altura del árbol de personas es: 4.
138>La persona con id 15 está en el árbol de personas.
139>La persona con mayor id en el árbol es: 
Persona 20:  Persona20, 20 años
Evento 20:  Descripcion20
Fecha: 2/2/20
140>Persona 5:  Persona5, 55 años
Evento 5:  Descripcion5
Fecha: 5/5/55
Persona 10:  Persona10, 10 años
Persona 15:  Persona15, 15 años
Evento 15:  Descripcion15
Fecha: 5/5/15
Persona 20:  Persona20, 20 años
Evento 20:  Descripcion20
Fecha: 2/2/20
141>142>La persona con id 10 NO está en el árbol de personas.
143>Persona 5:  Persona5, 55 años
Evento 5:  Descripcion5
Fecha: 5/5/55
Persona 15:  Persona15, 15 años
Evento 15:  Descripcion15
Fecha: 5/5/15
Persona 20:  Persona20, 20 años
Evento 20:  Descripcion20
Fecha: 2/2/20
144>145>El árbol de personas NO es perfecto.
146>147>148>149>150>151>152>El árbol de personas es perfecto.
153>154>155>156>157>158>159>160>161>162>163>164>165>166>167>168>169>170>171>172>173>174>175>176>177>178>179>La cantidad de personas en la lista es: 4.
180>Persona 0:  Persona0, 1 años
Evento 0:  Descripcion0
Fecha: 1/10/1100
Persona 1:  Persona1, 11 años
Evento 1:  Descripcion1
Fecha: 1/11/11
Persona 2:  Persona2, 22 años
Evento 2:  Descripcion2
Fecha: 2/22/22
Persona 3:  Persona3, 33 años
Evento 3:  Descripcion3
Fecha: 3/33/33
181>La persona con id 3 está en la lista.
182>La persona con id 5 NO está en la lista.
183>La persona con id 4 NO está en la lista.
184>La persona con id 0 está en la lista.
185>186>187>La cantidad de personas en la lista es: 3.
188>Persona 1:  Persona1, 11 años
Evento 1:  Descripcion1
Fecha: 1/11/11
Persona 2:  Persona2, 22 años
Evento 2:  Descripcion2
Fecha: 2/22/22
Persona 3:  Persona3, 33 años
Evento 3:  Descripcion3
Fecha: 3/33/33
189>190>Persona 2:  Persona2, 22 años
Evento 2:  Descripcion2
Fecha: 2/22/22
191>La cantidad de personas en la lista es: 2.
192>Persona 1:  Persona1, 11 años
Evento 1:  Descripcion1
Fecha: 1/11/11
Persona 2:  Persona2, 22 años
Evento 2:  Descripcion2
Fecha: 2/22/22
193>194>195>La cantidad de personas en la lista es: 0.
196>197>198>Persona 1: Juan, 20 años
Persona 2: Pedro, 21 años
Persona 3: Maria, 22 años
199>La cantidad de personas en la lista es: 3.
200>201>Fin.
